src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -253,9 +253,18 @@ snc_quota_manager::calc_node_quota_default() const {
 void snc_quota_manager::get_or_create_quota_context(
   std::unique_ptr<snc_quota_context>& ctx,
   std::optional<std::string_view> client_id,
+  const security::acl_principal* const principal,
```

Comment:
suggestion: Use a `std::optional` to make the comparison below simpler.

Reply:
`_sasl->principal()` returns a `const acl_principal&`.  Can it be a `std::optional<std::reference_wrapper<const acl_principal>>`?